parachain/pallets/ethereum-beacon-light-client/src/lib.rs

```diff
@@ -0,0 +1,710 @@
+				initial_sync.current_sync_committee_branch,
+				initial_sync.header.state_root,
+				CURRENT_SYNC_COMMITTEE_DEPTH,
+				CURRENT_SYNC_COMMITTEE_INDEX,
+			)?;
+
+			let period = Self::compute_current_sync_period(initial_sync.header.slot);
+
+			Self::store_sync_committee(period, initial_sync.current_sync_committee);
+
+			let block_root: H256 = merklization::hash_tree_root_beacon_header(initial_sync.header.clone())
+				.map_err(|_| DispatchError::Other("Header hash tree root failed"))?.into();
+
+			Self::store_header(block_root, initial_sync.header);
+
+			Self::store_genesis(Genesis { validators_root: initial_sync.validators_root });
+
+			Ok(())
+		}
+
+		fn process_sync_committee_period_update(
+			update: LightClientSyncCommitteePeriodUpdate,
+		) -> DispatchResult {
+			let sync_committee_bits = Self::convert_to_binary(update.sync_aggregate.sync_committee_bits.clone());
+
+			Self::sync_committee_participation_is_supermajority(sync_committee_bits.clone())?;
+
+			Self::verify_sync_committee(
+				update.next_sync_committee.clone(),
+				update.next_sync_committee_branch,
+				update.finalized_header.state_root,
+				NEXT_SYNC_COMMITTEE_DEPTH,
+				NEXT_SYNC_COMMITTEE_INDEX,
+			)?;
+
+			let block_root: H256 = merklization::hash_tree_root_beacon_header(update.finalized_header.clone())
+				.map_err(|_| DispatchError::Other("Header hash tree root failed"))?.into();
+
+			Self::verify_header(
+				block_root,
+				update.finality_branch,
+				update.attested_header.state_root,
+				FINALIZED_ROOT_DEPTH,
+				FINALIZED_ROOT_INDEX,
+			)?;
+
+			let current_period = Self::compute_current_sync_period(update.attested_header.slot);
+
+			Self::store_sync_committee(current_period + 1, update.next_sync_committee);
+
+			let sync_committee = <SyncCommittees<T>>::get(current_period);
+
+			let genesis = <ChainGenesis<T>>::get();
+
+			Self::verify_signed_header(
+				sync_committee_bits,
+				update.sync_aggregate.sync_committee_signature,
+				sync_committee.pubkeys,
+				update.fork_version,
+				update.attested_header,
+				genesis.validators_root,
+			)?;
+
+			Self::store_header(block_root, update.finalized_header);
+
+			Ok(())
+		}
+
+		fn process_finalized_header(update: LightClientFinalizedHeaderUpdate) -> DispatchResult {
+			let sync_committee_bits = Self::convert_to_binary(update.sync_aggregate.sync_committee_bits.clone());
+
+			Self::sync_committee_participation_is_supermajority(sync_committee_bits.clone())?;
+
+			let block_root: H256 = merklization::hash_tree_root_beacon_header(update.finalized_header.clone())
+				.map_err(|_| DispatchError::Other("Header hash tree root failed"))?.into();
+
+			Self::verify_header(
+				block_root,
+				update.finality_branch,
+				update.attested_header.state_root,
+				FINALIZED_ROOT_DEPTH,
+				FINALIZED_ROOT_INDEX,
+			)?;
+
+			let current_period = Self::compute_current_sync_period(update.attested_header.slot);
+
+			let sync_committee = <SyncCommittees<T>>::get(current_period);
+
```

Comment:
nit: Lets try and group statements in such a way that there isn't always an empty line between each and every statement. Makes it easier to understand how statements relate to each other. Of course in some contexts, the empty lines make sense, so its partially subjective style thing.

Reply:
Sure, I'll do that. I personally like formatting my code like because it creates some space that makes it easier for me to read, but but I'm happy to update it so it fits the rest of the codebase.